File: golangLeetcode/70.爬楼梯.go
/*
 * @lc app=leetcode.cn id=70 lang=golang
 *
 * [70] 爬楼梯
 */

// @lc code=start
func climbStairs(n int) int {
	if n < 1 {
		return 0
	}
	first := 0
	second := 1

	for step := 0; step < n; step++ {
		first, second = second, second+first
	}
	return second
}

// @lc code=end

File: LoveSwiftLeetcode/70.爬楼梯.swift
/*
 * @lc app=leetcode.cn id=70 lang=swift
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
    func climbStairs(_ n: Int) -> Int {
        var pre = 0
        var next = 1
        for _ in 0...n {
            (pre, next) = (next, pre + next)
        }
        return pre
    }
}
// @lc code=end

